tests/test_market_mechanisms.py

```diff
@@ -123,6 +99,170 @@ def test_market_mechanism():
     # return mr.all_orders, meta
 
 
+def test_complex_clearing():
```

Comment:
I would rather move these Tests to a file `test_market_mechanisms_complex` or `test_complex_market_mechanisms`